tests/test_counter.py

```diff
@@ -0,0 +1,109 @@
+"""This module tests the castanet.counter module."""
+
+from castanet import counter
+from castanet import generate_trees as generator
+
+
+def test_for_loops_dict():
+    """Check that for loops are counted correctly."""
+    directory = "./test_files"
+    file_list = generator.find_python_files(directory)
+    string_file_list = generator.read_files(directory, file_list)
+    tree_dict = generator.generate_cast(string_file_list)
+    for_dictionary = counter.count_forloops(tree_dict)
+    amount_for_loops =  counter.amount_loops(for_dictionary)
+
+    assert len(for_dictionary) == 5
+    assert amount_for_loops == 3
```

Comment:
One good testing practice is to keep your test input as close to the tests and their assertions as possible.

This way, someone looking at a test can make sense of the test just by looking at only the test. If the test contains the input, the action under test, and the assertion, then it's easy to see why the assertion was made.

In contrast, if the test input is far away from the test (i.e. in another file), then someone would have to look at two files to make sense of a test. Here, I'd also need to look at the files in the `test_files` directory to understand why the assertion that the number of `for` loops should be 3 makes sense. Moreover, if the test input is ever updated, it's not immediately clear which tests need to be updated as well.

This is not something that needs to be _fixed_, per se, but something to think about when approaching future testing. You may also consider refactoring some of the existing code to facilitate this practice.

For example, if `count_forloops` was rewritten so that it took in a string of code, rather than a dictionary of CSTs, you could write something like:

```python
def test_count_forloops_returns_number_of_forloops():
    input = '''
for _ in range(2):
    pass
# for
'''
    assert counter.count_forloops(input) == 1
```

By looking at only this test case, I can see exactly why the count should be 1. And if I were to update the input, then the assertion that also needs to be updated is directly below and easy to remember to change.

Reply:
Resolving until the next sprint.